bgpd/bgp_evpn.c

```diff
@@ -4576,16 +4576,32 @@ void bgp_evpn_advertise_type5_routes(struct bgp *bgp_vrf, afi_t afi,
 				/* apply the route-map */
 				if (bgp_vrf->adv_cmd_rmap[afi][safi].map) {
 					route_map_result_t ret;
+					struct bgp_path_info tmp_pi;
+					struct bgp_path_info_extra tmp_pie;
+					struct attr tmp_attr;
+
+					tmp_attr = *pi->attr;
+
+					/* Fill temp path_info */
+					prep_for_rmap_apply(
+						&tmp_pi, &tmp_pie, rn, pi,
+						pi->peer, &tmp_attr);
+
+					RESET_FLAG(tmp_attr.rmap_change_flags);
 
 					ret = route_map_apply(
 						bgp_vrf->adv_cmd_rmap[afi][safi]
 							.map,
-						&rn->p, RMAP_BGP, pi);
+						&rn->p, RMAP_BGP, &tmp_pi);
 					if (ret == RMAP_DENYMATCH)
```

Comment:
I think we should do here `bgp_attr_flush(&tmp_pi);`, no?

Reply:
I took a closer look, and you are quite right. With 'next' and 'goto' clauses etc., some 'sets' might've executed before eventually returning DENYMATCH. I'll fix this. Good catch!